### PR TITLE
Enable SNI based on compile option.

### PR DESCRIPTION
**Pull request by tajnisha**
SNI support is added in this patch to handle
a customer request. 

If this is enabled, the packets from client
will also have the SNI(server name indication)
extension that contains the server name.

### CODE REVIEW

File: network/network_if_linux.c [hunk 1]
```diff
@@ -404,7 +396,50 @@ int fdo_curl_setup(fdo_ip_address_t *ip_addr, uint16_t port, bool tls)
...
+				LOG(LOG_ERROR, "Snprintf() failed!\n");
+				goto err;
+			}
+			if (strcat_s(url, HTTP_MAX_URL_SIZE, temp) != 0) {
+				LOG(LOG_ERROR, "Strcat() failed!\n");
+				goto err;
+			}
+		}
 
+		LOG(LOG_INFO, "nisha, the URL: %s\n",url);
```
Comment (shrikant1407) [inline network/network_if_linux.c:442]: remove this log

File: network/network_if_linux.c [hunk 2]
```diff
@@ -404,7 +396,50 @@ int fdo_curl_setup(fdo_ip_address_t *ip_addr, uint16_t port, bool tls)
...
+			}
+			if (strcat_s(temp, 2*HTTP_MAX_URL_SIZE, ":") != 0) {
+				LOG(LOG_ERROR, "Strcat() failed!\n");
+				goto err;
+			}
+			if (strcat_s(temp, 2*HTTP_MAX_URL_SIZE, ip_ascii) != 0) {
+				LOG(LOG_ERROR, "Strcat() failed!\n");
+				goto err;
+			}
+			LOG(LOG_INFO, "nisha, the temp: %s\n",temp);
```
[the remark this answers is not in the document]
Comment (tajnisha) [reply network/network_if_linux.c:422]: sure

File: cmake/cli_input.cmake [hunk 3]
```diff
@@ -11,11 +11,12 @@ set (CSTD c99)
 set (TLS openssl)
 set (DA ecdsa384)
 set (AES_MODE gcm)
-set (BUILD release)
+set (BUILD debug)
```
[the remark this answers is not in the document]
Comment (tajnisha) [reply cmake/cli_input.cmake:14]: sure

File: network/network_if_linux.c (hunk 1 above)
Comment (tajnisha) [reply network/network_if_linux.c:442]: sure

**Review by shrikant1407 — APPROVED** [review]

**Review by v-vasavi — APPROVED** [review]

**Pull request merged**